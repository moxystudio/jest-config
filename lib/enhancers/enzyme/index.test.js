'use strict';

const { withEnzymeWeb, withEnzymeReactNative } = require('.');

const buildConfig = () => ({
    testEnvironment: 'jsdom',
    globals: {
        foo: 'bar',
    },
    setupFilesAfterEnv: ['foo'],
});

describe('withEnzymeWeb', () => {
    it('should match snapshot', () => {
        expect(withEnzymeWeb('enzyme-adapter-react-16')(buildConfig())).toMatchSnapshot();
    });

    it('should throw if no adapter is passed', () => {
        expect(() => withEnzymeWeb()).toThrow(/please provide an enzyme adapter/i);
    });

    it('should throw if testEnvironment is not jsdom', () => {
        const config = {
            ...buildConfig(),
            testEnvironment: 'node',
        };

        expect(() => withEnzymeWeb('enzyme-adapter-react-16', 'web')(config)).toThrow(/expected testEnvironment to be set to "jsdom"/i);
    });
});

describe('withEnzymeReactNative', () => {
    it('should match snapshot', () => {
        expect(withEnzymeReactNative('enzyme-adapter-react-16')(buildConfig())).toMatchSnapshot();
    });

    it('should throw if no adapter is passed', () => {
        expect(() => withEnzymeReactNative()).toThrow(/please provide an enzyme adapter/i);
    });

    it('should throw if testEnvironment is not jsdom', () => {
        const config = {
            ...buildConfig(),
            testEnvironment: 'node',
        };

        expect(() => withEnzymeReactNative('enzyme-adapter-react-16')(config)).toThrow(/expected testEnvironment to be set to "jsdom"/i);
    });
});
