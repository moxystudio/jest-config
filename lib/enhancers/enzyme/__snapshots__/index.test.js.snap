// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withEnzymeReactNative should match snapshot 1`] = `
Object {
  "globals": Object {
    "__MOXY_JEST_CONFIG_ENZYME_ADAPTER__": "enzyme-adapter-react-16",
    "foo": "bar",
  },
  "setupFilesAfterEnv": Array [
    "foo",
    "<PROJECT_ROOT>/lib/enhancers/enzyme/setup.js",
    "<PROJECT_ROOT>/lib/enhancers/enzyme/setup-react-native.js",
  ],
  "testEnvironment": "jsdom",
}
`;

exports[`withEnzymeWeb should match snapshot 1`] = `
Object {
  "globals": Object {
    "__MOXY_JEST_CONFIG_ENZYME_ADAPTER__": "enzyme-adapter-react-16",
    "foo": "bar",
  },
  "setupFilesAfterEnv": Array [
    "foo",
    "<PROJECT_ROOT>/lib/enhancers/enzyme/setup.js",
    "<PROJECT_ROOT>/lib/enhancers/enzyme/setup-web.js",
  ],
  "testEnvironment": "jsdom",
}
`;
